pkg/analysis/passes/checksum/checksum.go

```diff
@@ -0,0 +1,155 @@
+package checksum
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/grafana/plugin-validator/pkg/analysis"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/readme"
+	"github.com/grafana/plugin-validator/pkg/logme"
+)
+
+var (
+	checksumInvalid     = &analysis.Rule{Name: "checksum-invalid", Severity: analysis.Error}
+	checksumFormatError = &analysis.Rule{Name: "checksum-format-error", Severity: analysis.Error}
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name:     "checksum",
+	Requires: []*analysis.Analyzer{readme.Analyzer},
+	Run:      run,
+	Rules:    []*analysis.Rule{checksumInvalid},
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+
+	checksum := pass.CheckParams.Checksum
+
+	// skip if no provided
+	if checksum == "" || len(checksum) == 0 {
+		return nil, nil
+	}
+
+	checksum, err := resolveCheckSum(strings.TrimSpace(checksum))
+	if err != nil {
+		pass.ReportResult(
+			pass.AnalyzerName,
+			checksumFormatError,
+			fmt.Sprintf("%s: %s", err.Error(), pass.CheckParams.Checksum),
+			"Make sure you provide a valid checksum as a string or a valid URL to a checksum. Only MD5 and SHA1 checksums are supported.",
+		)
+		return nil, nil
+	}
+
+	checkSumType := ""
+
+	// determine if it is md5 or sha1
+	if len(checksum) == 32 {
+		checkSumType = "md5"
+	} else if len(checksum) == 40 {
+		checkSumType = "sha1"
+	} else {
+		pass.ReportResult(
+			pass.AnalyzerName,
+			checksumFormatError,
+			fmt.Sprintf("Invalid checksum format: %s", checksum),
+			"Make sure you provide a valid checksum as a string or a valid URL to a checksum. Only MD5 and SHA1 checksums are supported.",
+		)
+		return nil, nil
+	}
+	isError := false
+
+	switch checkSumType {
+	case "md5":
+		if checksum != pass.CheckParams.ArchiveCalculatedMD5 {
+			isError = true
+		}
+	case "sha1":
+		if checksum != pass.CheckParams.ArchiveCalculatedSHA1 {
+			isError = true
+		}
+	}
+
+	if isError {
+		pass.ReportResult(
+			pass.AnalyzerName,
+			checksumInvalid,
+			fmt.Sprintf("The provided checksum %s does not match the plugin archive", checksum),
+			"The plugin archive does not match the provided checksum in the submission form. Please double check you are providing the correct plugin archive and checksum.",
+		)
+	}
+
+	return nil, nil
+}
+
+// sums can come as:
+// 04b2f189616b65985d219ca094f3609b  arduino-cli.yaml
+// 0b41e5bffba89764a494f420347fe5110fa2886c  arduino-cli.yaml
+// 0b41e5bffba89764a494f420347fe5110fa2886c
+// 04b2f189616b65985d219ca094f3609b
+// we only care about the first part
+func sanitizeCheckSum(checksum string) string {
+	return strings.Split(checksum, " ")[0]
```

Comment:
Can they also be separated by other whitespaces (like tabs) or just spaces? In case there can also be other whitespace characters, you can use [strings.Fields](https://pkg.go.dev/strings#Fields) instead of strings.Split with a space

Reply:
As far as I know it can be only whitespaces but I like the idea of using something that can handle them all.